File: fromJAVAtoPYTHON/04_OOP/06.Constructors_in_Python/Country.py
class Country:
    # constructor
    def __init__(self) -> None:
        print('constructor1')
    
    def __init__(self, name='Default') -> None:
        self.name = name
        print('constructor2')
        print(name)

    def instance_method(self):
        print('instance method')

#instances
india1 = Country()
india2 = Country('India')
india1.instance_method()

File: fromJAVAtoPYTHON/04_OOP/06.Constructors_in_Python/Country.java
public class Country {
    Country() {
        System.out.println("constructor1");
    }

    Country(String name) {
        System.out.println("constructor2");
    }

    void instance_method() {
        System.out.println("instance method");
    }
    public static void main(String[] args) {
        Country india1 = new Country();
        Country india2 = new Country("india");
        india1.instance_method();
    }
}